specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args)
+                    -> HRESULT {
+                    wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+                    CHECK_FAILURE(args->get_Frame(&webviewFrame));
+                    wil::com_ptr<ICoreWebView2ExperimentalFrame> frameExperimental =
+                        webviewFrame.try_query<ICoreWebView2ExperimentalFrame>();
+                    if (frameExperimental)
+                    {
+                        frameExperimental->add_DOMContentLoaded(
+                            Callback<
+                                ICoreWebView2ExperimentalFrameDOMContentLoadedEventHandler>(
+                                [](
+                                    ICoreWebView2Frame* frame,
+                                    ICoreWebView2DOMContentLoadedEventArgs* args) -> HRESULT {
+                                    wil::com_ptr<ICoreWebView2ExperimentalFrame> frameExperimental;
+                                    frame->QueryInterface(IID_PPV_ARGS(&frameExperimental));
+
+                                    frameExperimental->ExecuteScript(
+                                        LR"~(
+                                        let content = document.createElement("h2");
+                                        content.style.color = 'blue';
+                                        content.textContent = "The url of the iframe is " + window.location.href.toString();
+                                        document.body.appendChild(content);
+                                        )~",
+                                        Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                                            [](HRESULT error, PCWSTR result) -> HRESULT {
+                                                return S_OK;
+                                            })
+                                            .Get());
+                                    return S_OK;
+                                })
+                                .Get(),
+                            NULL);
+                    }
+                    return S_OK;
+                })
+                .Get(),
+            NULL));
+    }
+```
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## IDL
+```c#
+/// WebView2Frame provides direct access to the iframes information and use.
+[uuid(423ba05f-55dd-4c50-8339-22afd009ed31), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalFrame : IUnknown {
+  /// Add an event handler for the `NavigationStarting` event.
+  /// A frame navigation will raise a `NavigationStarting` event and
+  /// a `CoreWebView2.FrameNavigationStarting` event. All of the
+  /// `FrameNavigationStarting` event handlers will be run before the
+  /// `NavigationStarting` event handlers. All of the event handlers share a
```

Comment:
Why this event order? I was expecting both events to raise for one frame before moving on to the next frame.

Actually, the fact that these can be ordered like this makes it sound like the events are async?

Reply:
Update docs to make it clear this is all of the blah event handlers for the frame that is navigating, (not all the event handlers across multiple frames)